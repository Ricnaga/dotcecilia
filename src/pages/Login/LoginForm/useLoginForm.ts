import { useFormik } from 'formik';
import { useNavigate } from 'react-router-dom';
import { CALCULATOR } from '@application/routes/routes';
import { ENV_PASSWORD, ENV_USER } from '@config';
import { useLocalStorage } from '@shared/hooks/useLocalStorage';
import { FormFields } from '.';

type FormikValues = Record<'name' | 'password', string>;

export const useLoginForm = () => {
  const { saveUserData } = useLocalStorage();
  const navigate = useNavigate();
  const initialValues = {
    [FormFields.name]: '',
    [FormFields.password]: '',
  };

  const validateCredentials = (values: FormikValues) =>
    [ENV_USER].includes(values.name.toLowerCase()) &&
    [ENV_PASSWORD].includes(values.password);

  const onSubmit = (values: FormikValues) => {
    if (validateCredentials(values)) {
      saveUserData(values);
      navigate(CALCULATOR);
    }
  };

  const formik = useFormik({
    initialValues,
    onSubmit,
  });

  return {
    data: {
      values: formik.values,
    },
    functions: {
      onChange: formik.handleChange,
      onSubmit: formik.handleSubmit,
    },
  };
};
