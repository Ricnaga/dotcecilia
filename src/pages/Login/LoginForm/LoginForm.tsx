import { CeciliaButton, CeciliaText } from '@shared/components';
import { useLoginForm } from './hooks/useLoginForm';

export enum FormFields {
  name = 'name',
  password = 'password',
}

export function LoginForm() {
  const {
    data: { values },
    functions: { onChange, onSubmit },
  } = useLoginForm();

  return (
    <form onSubmit={onSubmit}>
      <div>
        <CeciliaText
          placeholder="Digite o usuÃ¡rio"
          value={values[FormFields.name]}
          onChange={onChange(FormFields.name)}
        />
      </div>

      <div>
        <CeciliaText
          placeholder="Digite a senha"
          type="password"
          value={values[FormFields.password]}
          onChange={onChange(FormFields.password)}
        />
      </div>

      <CeciliaButton title="Entrar" type="submit" />
    </form>
  );
}
