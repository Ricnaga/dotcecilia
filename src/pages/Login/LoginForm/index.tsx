import { CeciliaButton, CeciliaInput } from '../../../shared/components';
import { useLoginForm } from './useLoginForm';

export enum FormFields {
  name = 'name',
  password = 'password',
}

export function LoginForm() {
  const {
    data: { values },
    functions: { onChange, onSubmit },
  } = useLoginForm();

  return (
    <form onSubmit={onSubmit}>
      <div>
        <CeciliaInput
          placeholder="Digite o usuÃ¡rio"
          value={values[FormFields.name]}
          onChange={onChange(FormFields.name)}
        />
      </div>

      <div>
        <CeciliaInput
          placeholder="Digite a senha"
          type="password"
          value={values[FormFields.password]}
          onChange={onChange(FormFields.password)}
        />
      </div>

      <CeciliaButton title="Entrar" type="submit" />
    </form>
  );
}
