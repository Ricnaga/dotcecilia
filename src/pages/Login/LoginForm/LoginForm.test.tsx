import { render } from '@testing-library/react';
import { LoginForm } from './LoginForm';

jest.mock('./hooks/useLoginForm', () => ({
  __esModule: true,
  useLoginForm: () => ({
    data: {
      values: {
        mockName: 'mockName',
        mockPassword: 'mockPassword',
      },
    },
    functions: {
      onChange: jest.fn(),
      onSubmit: jest.fn(),
    },
  }),
}));

describe('Component: LoginForm ', () => {
  it('should render correctly', () => {
    const { container } = render(<LoginForm />);
    expect(container).toMatchSnapshot();
  });
});
