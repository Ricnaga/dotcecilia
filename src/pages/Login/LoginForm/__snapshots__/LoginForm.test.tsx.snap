// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LoginForm  should render correctly 1`] = `
<div>
  <form>
    <div>
      <label
        class="text-xl"
        for="text"
      >
        <input
          placeholder="Digite o usuÃ¡rio"
          type="text"
          value=""
        />
      </label>
    </div>
    <div>
      <label
        class="text-xl"
        for="text"
      >
        <input
          placeholder="Digite a senha"
          type="password"
          value=""
        />
      </label>
    </div>
    <button
      class="btn btn-yellow undefined"
      type="submit"
    >
      Entrar
    </button>
  </form>
</div>
`;
